iroh-relay/src/server/client.rs

```diff
@@ -557,6 +567,41 @@ impl Sink<Frame> for RateLimitedRelayedStream {
     }
 }
 
+/// Counts how many `NodeId`s seen, how many times.
+/// Gets reset every day.
+#[derive(Debug)]
+struct ClientCounter {
+    clients: HashSet<NodeId>,
+    last_clear_date: Date,
+}
+
+impl Default for ClientCounter {
+    fn default() -> Self {
+        Self {
+            clients: HashSet::new(),
+            last_clear_date: OffsetDateTime::now_utc().date(),
+        }
+    }
+}
+
+impl ClientCounter {
+    fn check_and_clear(&mut self) {
+        let today = OffsetDateTime::now_utc().date();
+        if today != self.last_clear_date {
+            self.clients.clear();
+            self.last_clear_date = today;
+        }
+    }
+
+    /// Updates the client counter.
+    fn update(&mut self, client: NodeId) -> bool {
+        self.check_and_clear();
+        let new_conn = !self.clients.contains(&client);
+        self.clients.insert(client);
+        new_conn
```

Comment:
You're again doing two set lookups where only one is needed I think:
```suggestion
        self.clients.insert(client);
```

Also, documenting the return value wouldn't do any harm :)